### PR TITLE
Sign in sign up auth

### PR DESCRIPTION
**Pull request by ark2002**
- Added signin page
- Added signup page
- added authentication

Preview:

https://deploy-preview-2--get-it-noted.netlify.app/

recording

### CODE REVIEW

**Review by jeetbhalani15 — APPROVED** [review]
-API class works fine
-Naming looks good
-Readable code

File: src/backend/db/users.js [hunk 1]
```diff
@@ -8,10 +8,10 @@ import { formatDate } from "../utils/authUtils";
 export const users = [
   {
     _id: uuid(),
```
Comment (sahiba0915) [inline src/backend/db/users.js:10]: why this _id? can use simple key name right?

File: src/frontend/context/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, { useState, useContext, createContext } from "react"
+
+const AuthContext = createContext();
+
+
```
Comment (sahiba0915) [inline src/frontend/context/auth-context.js:5]: remove unwanted spaces

File: src/frontend/pages/SignInPage/SignInPage.css [hunk 3]
```diff
@@ -0,0 +1,50 @@
+.login__form {
+    gap: 1rem;
+    width: 30%;
+    margin: 10rem auto;
+    padding: 2rem 5rem;
+    box-shadow: #0000003d 0px 3px 8px;
```
Comment (sahiba0915) [inline src/frontend/pages/SignInPage/SignInPage.css:6]: can use variables

**Review by sahiba0915 — APPROVED** [review]
Just added one suggestion rest looks good, properly structured, code looks clean.

**Pull request merged**